src/compiler/utilities.ts

```diff
@@ -6292,8 +6292,18 @@ namespace ts {
 
     /** Returns true for the first identifier of 1) an `implements` clause, and 2) an `extends` clause of an interface. */
     function isFirstIdentifierOfNonEmittingHeritageClause(node: Node): boolean {
-        // Number of parents to climb from identifier is 2 for `implements I`, 3 for `implements x.I`

```

Comment:
At the time I was thinking that the left side of a property access was the _top_ of the tree 🤦‍♂